api/src/handlers/reports_staking.rs

```diff
@@ -38,34 +39,54 @@ fn filter_and_sort_stakes(records: &mut Vec<StakingChange>) {
 }
 
 async fn get_planned_stakes(context: WrappedContext) -> anyhow::Result<Vec<StakingChange>> {
+    let psql_client = &context.read().await.psql_client;
     let cache = &context.read().await.cache;
     let mut records = Vec::new();
+    let last_epoch = match get_last_epoch(psql_client).await? {
+        Some(last_epoch) => last_epoch,
+        _ => return Ok(Default::default()),
+    };
 
     let CachedScores { scores, .. } = cache.get_validators_scores();
-    let stakes = cache.get_validators_current_stakes();
+    let validators = cache.get_validators();
```

Comment:
combine the row with the `context.read().await` - so the context lock is dropped as soon as possible

Reply:
same above for `psql_client`